integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/common/ServerConfigurationManager.java

```diff
@@ -298,16 +292,21 @@ public void applyConfiguration(File newConfig) throws AutomationUtilException, I
     }
 
     /**
-     * apply configuration file and restart micro integrator server to take effect the configuration
+     * apply configuration file and optionally restart micro integrator server to take effect the configuration
      *
-     * @param newConfig configuration file
+     * @param newConfig       configuration file
+     * @param restartRequired optional Restart flag
      * @throws AutomationUtilException - throws if apply configuration fails
      * @throws IOException             - throws if apply configuration fails
      */
-    public void applyMIConfiguration(File newConfig) throws AutomationUtilException, IOException {
+    public void applyMIConfiguration(File newConfig, boolean restartRequired)
+            throws AutomationUtilException, IOException {
         //to backup existing configuration
         applyConfigurationUtil(newConfig, newConfig);
-        restartMicroIntegrator();
+        if (restartRequired) {
```

Comment:
This is a compile time decision. Isn't it? We can have a separate method in such a scenario to improve the readability and additionally save CPU cycles. Just sharing my thoughts while reading the code. :)

Reply:
Fixed it. https://github.com/wso2/micro-integrator/pull/456